assets/js/components/ExchangeRates.js

```diff
@@ -1,106 +1,175 @@
 import axios from 'axios';
 
-class ExchangeRates extends Component {
-    constructor() {
-        super();
-        this.state = {
-            rates: {},
-            loading: true,
-            date: (new Date()).toISOString().split('T')[0]
-        };
-    }
+const ExchangeRates = () => {
+    const [rates, setRates] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [date, setDate] = useState((new Date()).toISOString().split('T')[0]);
+    const [calculateValue, setCalculateValue] = useState('');
+    const [calculateBuy, setCalculateBuy] = useState(-1);
+    const [calculateSell, setCalculateSell] = useState(-1);
+    const [calculateExchange, setCalculateExchange] = useState('');
+    const [calculateCode, setCalculateCode] = useState('');
 
-    getBaseUrl() {
-        return 'http://telemedi-zadanie.localhost';
-    }
+    const getBaseUrl = () => 'http://telemedi-zadanie.localhost';
 
-    componentDidMount() {
+    useEffect(() => {
         const urlParams = new URLSearchParams(window.location.search);
-        const date = urlParams.get('date')
-        if (date !== null) {
-            this.state.date = date
+        const dateParam = urlParams.get('date');
+        if (dateParam !== null) {
+            setDate(dateParam);
         }
+        fetchExchangeRates();
+    }, []);
 
-        this.fetchExchangeRates();
-    }
+    useEffect(() => {
+        recalculate();
+        recalculateCode();
+    }, [calculateValue, calculateBuy, calculateSell, rates]);
 
-    fetchExchangeRates() {
-        const baseUrl = this.getBaseUrl();
-        const date = this.state.date;
-        axios.get( `${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
-            this.setState({
-                rates: response.data,
-                loading: false
-            })
-        }).catch(function (error) {
+    const fetchExchangeRates = () => {
+        const baseUrl = getBaseUrl();
+        axios.get(`${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
+            setRates(response.data);
+            setLoading(false);
+        }).catch(error => {
             console.error(error);
-            this.setState({ rates: false, loading: false});
+            setRates(false);
+            setLoading(false);
         });
-    }
+    };
 
-    changeDate = (event) => {
+    const changeDate = (event) => {
         const formattedDate = event.target.value;
-        this.setState({date: formattedDate})
+        setDate(formattedDate);
         window.history.replaceState(null, null, `?date=${formattedDate}`);
-        this.fetchExchangeRates()
-    }
+        fetchExchangeRates();
+    };
+
+    const setExchangeValue = (event) => {
+        setCalculateValue(event.target.value);
+    };
+
+    const setExchangeSell = (event) => {
+        setCalculateSell(event.target.value);
+    };
+
+    const setExchangeBuy = (event) => {
+        setCalculateBuy(event.target.value);
+    };
```

Comment:
Troche nazwinictwo do funkcji mi nie pasuje (ale pewnie się tylko czepiam). Bo to nie są funkcje która po prostu ustawiają wartość ale ustawiją ją z `event`'u. Może coś w tym guście: `setExchangeValueFromEvent`?